messaging/connectors/kafka/src/main/java/io/helidon/messaging/connectors/kafka/PartitionsAssignedLatch.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.messaging.connectors.kafka;
+
+import java.util.Collection;
+import java.util.concurrent.CountDownLatch;
+import java.util.logging.Logger;
+
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Waiting latch for partition assigment, after that is consumer ready to receive.
+ */
+class PartitionsAssignedLatch extends CountDownLatch implements ConsumerRebalanceListener {
```

Comment:
This seems not to do anything useful. Isn't the contract of `ConsumerRebalancerListener` a bit more complicated?

Reply:
From the point of view of reading new events, we don't care about the partitions because we don't specify them.
But this could affects pending commits. This is the current strategy when this happens:
- Event is coming from partition A.
- User is processing it.
- Kafka revoke that partition.
- User ack, commit is sent.
- Kafka should throw CommitFailedException, because that partition doesn't exist.
- The publisher will report a failure.

It could be possible to remove temporarily from pending commits those that are related to the revoked partition. And add them again to the pending commits when that partition is back. But this could never happen, so we would need to take care about expiring it at some point.

It seems that ConsumerRebalancerListener is triggered in the same thread that is polling from Kafka. That makes the things much easier in order to modify the pending commits.

For the time being I would keep it as it is.